authors/apps/profiles/tests/test_following.py

```diff
@@ -0,0 +1,145 @@
+        self.response_message_test(msesage, response)
+
+    def test_get_following_without_auth(self):
+        response = self.client.get(
+            self.following_url(), format='json')
+        self.forbidden_403(response)
+        msesage = 'Authentication credentials were not provided.'
+        self.response_message_test(msesage, response)
+
+    def test_get_following_with_auth(self):
+        self.follow(self.follow_existing_user_url())
+        token = self.login_user2()
+        response = self.client.get(
+            self.following_url(),
+            HTTP_AUTHORIZATION='Token ' + token, format='json')
+        msesage = '"followers_count":1'
+        self.response_message_test(msesage, response)
+        self.ok_200(response)
+
+    def test_get_followers_with_auth(self):
+        self.follow(self.follow_existing_user_url())
+        token = self.login_user2()
+        response = self.client.get(
+            self.followers_url(),
+            HTTP_AUTHORIZATION='Token ' + token, format='json')
+        msesage = '"following_count":1'
+        self.response_message_test(msesage, response)
+        self.ok_200(response)
+
+    def test_follow_with_auth(self):
+        response = self.follow(self.follow_existing_user_url())
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        msesage = '"followers_count":1'
+        self.response_message_test(msesage, response)
+
+    def test_follow_without_auth(self):
+        self.signup_user()
+        self.login_user2()
+        url = reverse("profiles:follow", kwargs={
+            "username": self.test_user_2['user']['username']})
+        response = self.client.post(url, format='json')
+        self.not_found_404(response)
+        msesage = 'You must be logged in first'
+        self.response_message_test(msesage, response)
+
+    def test_unfollow_with_auth(self):
+        self.follow(self.follow_existing_user_url())
+        self.signup_user()
+        self.test_user['user']['username'] = 'testuser'
+        token = self.login_user().data['token']
+        response = self.client.delete(
+            self.follow_existing_user_url(),
+            HTTP_AUTHORIZATION='Token ' + token, format='json')
+        self.ok_200(response)
+        msesage = '"followers_count":0'
+        self.response_message_test(msesage, response)
+
+    def test_unfollow_without_auth(self):
+        self.follow(self.follow_existing_user_url())
+        response = self.client.delete(
+            self.follow_existing_user_url(), format='json')
+        self.not_found_404(response)
+        msesage = 'You must be logged in first'
+        self.response_message_test(msesage, response)
+
+    def test_follow_non_existing_user(self):
+        response = self.follow(self.follow_non_existing_user_url())
+        self.not_found_404(response)
+        msesage = 'You are trying to follow a user who doe not exist, please check the username again'
```

Comment:
@Hesbon5600 "Doe not exist" A typo here.

Reply:
Fixed